os/include/tinyara/pm/pm.h

```diff
@@ -88,6 +88,24 @@
 #include <queue.h>
 #include <semaphore.h>
 
+#if !defined(CONFIG_PM_NDOMAINS)
+#define CONFIG_PM_NDOMAINS 31
+#endif
+
+/* This enumeration provides all power management domains */
+enum pm_domain_e {
+	PM_IDLE_DOMAIN 	= 0,
+#ifdef CONFIG_LCD
+	PM_LCD_DOMAIN 	= 1,
+#endif
+	PM_UART_DOMAIN 	= 2,
+	PM_WIFI_DOMAIN 	= 3,
```

Comment:
I have a question, from app side, once the PM_WIFI_DOMAIN lockcount is 0, can we understand it as there will no be wifi operation ongoing as of that period of time?

Reply:
Yes, or in other words you can say "Wifi is ready to go into PM_SLEEP". If it wants to perform some task then it must suspend the PM using PMIOC_SUSPEND